### PR TITLE
Facebook authentification plugin v2.0

### PR DESCRIPTION
**Pull request by baelmyhu**
Facebook authentification update
https://support.chamilo.org/issues/7605

### CODE REVIEW

File: main/auth/external_login/facebook.inc.php [hunk 1]
```diff
@@ -11,36 +11,61 @@
...
+	   // graph api request for user data
+	    $request = new FacebookRequest( $session, 'GET', '/me' );
+	    $response = $request->execute();
+   	    // get response
+           $graphObject = $response->getGraphObject(); 
+           $username = change_to_valid_chamilo_login($graphObject->getProperty('email'));
+           $email = $graphObject->getProperty('email');
+	    $locale = $graphObject->getProperty('locale');
+	    $language = get_language($locale);
+	    if(!$language)$language='pl_PL';
```
Comment (ywarnier) [inline main/auth/external_login/facebook.inc.php]: That's probably not a good idea (apart from the fact that spacing is wrong and that you should use brackets {}, the default language should be en_US

File: main/auth/external_login/facebook.inc.php [hunk 2]
```diff
@@ -73,26 +98,24 @@ function facebook_connect() {
...
-    global $facebook, $facebook_config;
 
-    $login_url = $facebook->getLoginUrl(
-            array(
-                'scope' => 'email,publish_stream',
-                'redirect_uri' => $facebook_config['return_url']
-            )
-    );
-    return $login_url;
+	 global $facebook_config;
```
Comment (ywarnier) [inline main/auth/external_login/facebook.inc.php]: One space too much here

File: main/auth/external_login/facebook.inc.php [hunk 3]
```diff
@@ -73,26 +98,24 @@ function facebook_connect() {
...
+	$helper = new FacebookRedirectLoginHelper($facebook_config['return_url']);
+ 	$login_url =   $helper->getLoginUrl(array(
+                'scope' => 'email));
```
Comment (ywarnier) [inline main/auth/external_login/facebook.inc.php]: Missing a single quote after "email"

File: main/auth/external_login/facebook.inc.php [hunk 4]
```diff
@@ -104,3 +127,20 @@ function change_to_valid_chamilo_login($in_txt) {
     return preg_replace("/[^a-zA-Z1-9_\-.]/", "_", $in_txt);
     exit;
 }
+
+/**
+ * Get user language
+ * @input : a string
+ *  @return : a string containing valid language
```
Comment (ywarnier) [inline main/auth/external_login/facebook.inc.php]: Wrong spacing

File: main/auth/external_login/facebook.inc.php [hunk 5]
```diff
@@ -104,3 +127,20 @@ function change_to_valid_chamilo_login($in_txt) {
...
+ * */
```
Comment (ywarnier) [inline main/auth/external_login/facebook.inc.php]: Wrong spacing (only one "*" needed)

File: main/auth/external_login/facebook.inc.php [hunk 6]
```diff
@@ -11,36 +11,61 @@
...
+use Facebook\HttpClients\FacebookHttpable;
+
 require_once dirname(__FILE__) . '/functions.inc.php';
 
+FacebookSession::setDefaultApplication( $facebook_config['appId'],$facebook_config['secret']);
 /**
  * This function connect to facebook and retrieves the user info
  * If user does not exist in chamilo, it creates it and logs in
  * If user already exists, it updates his info
  * */
```
Comment (ywarnier) [inline main/auth/external_login/facebook.inc.php]: Only one "*" needed

File: main/auth/external_login/facebook.inc.php [hunk 7]
```diff
@@ -104,3 +127,20 @@ function change_to_valid_chamilo_login($in_txt) {
...
     exit;
 }
+
+/**
+ * Get user language
+ * @input : a string
+ *  @return : a string containing valid language
+ * */
+
+function get_language($language = 'pl_PL') {
```
Comment (ywarnier) [inline main/auth/external_login/facebook.inc.php]: Default should be en_US, bracket should be on the next line. Spacing is bad for the following lines

File: main/auth/external_login/facebook.init.php [hunk 8]
```diff
@@ -14,19 +14,14 @@
...
-global $facebook;
-$facebook = new Facebook(array(
-            'appId' => $facebook_config['appId'],
-            'secret' => $facebook_config['secret']
-        ));
-require_once dirname(__FILE__) . '/facebook.inc.php';
\ No newline at end of file
+
+require_once dirname(__FILE__) . '/../../inc/conf/auth.conf.php';
+
```
Comment (ywarnier) [inline main/auth/external_login/facebook.init.php]: The facebook button below should be in English by default

**Comment by nicogaldo** [comment]
Some progress? I can not run in Chamilo 1.9.10.2

**Comment by ywarnier** [comment]
This patch was sent after the publication of 1.9.10.2. You should apply the patch or update your portal to 1.9.x development branch (proceed with caution, taking backups).

**Comment by nicogaldo** [comment]
I have updated and followed all the steps. When I click to Facebook icon on the login page Chamilo It Takes to be the Facebook Login eleven logged into facebook, it redirects the user back to Chamilo but the user is not logged in nor registered. Any ideas?

**Pull request closed without merging**